### PR TITLE
paypal express checkout

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ScottDowne** [comment]
@alicoding r?

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ $> cp sample.env .env
 $> npm start
 ```
 
+<<<<<<< HEAD
```
Comment (alicoding) [inline README.md]: Remove this

File: server.js [hunk 2]
```diff
@@ -3,7 +3,9 @@ var Path = require('path');
 
 var Hapi = require('hapi');
 var Good = require('good');
+
```
Comment (alicoding) [inline server.js]: remove this extra line

File: server.js [hunk 3]
```diff
@@ -71,6 +73,165 @@ server.route([
...
+            INITAMT: data.AMT,
+            CURRENCYCODE: data.CURRENCYCODE
+          }
+        }, function(err, httpResponse, body) {
+          if (err) {
+            return console.error('donation failed:', err);
+          }
+          var bodyData = querystring.parse(body);
+          // The below TRANSACTIONID is undefined.
+          // I'm not sure how to get this yet for recurring payments.
```
Comment (alicoding) [inline server.js]: have you file a bug for this?

File: server.js [hunk 4]
```diff
@@ -71,6 +73,165 @@ server.route([
...
+          reply.redirect('/thank-you/?tx=' + bodyData.PAYMENTINFO_0_TRANSACTIONID + '&amt=' + data.AMT + '&cc=' + data.CURRENCYCODE);
+        });
+      });
+    }
+  }, {
+    method: 'POST',
+    path: '/paypal-one-time',
+    handler: function(request, reply) {
+      var transaction = request.payload;
```
Comment (alicoding) [inline server.js]: what would happen if you pass in an empty payload here? would it fail? I think we should check and validate a bit to protect our server from crashing?

**Comment by alicoding** [comment]
Left some comments for you to answer.

File: server.js (hunk 3 above)
Comment (ScottDowne) [inline server.js]: This is mostly a question for adam.

I need to know how metrics on recurring payments is handled before I figure out how to get the id to pass along to metrics.

Comment (ScottDowne) [inline server.js]: https://github.com/mozilla/donate.mozilla.org/issues/119

It's kinda low priority compared to other issues.

**Comment by ScottDowne** [comment]
Updated, r?

**Comment by alicoding** [comment]
R+

**Pull request merged**